tensorflow/core/ops/string_ops.cc

```diff
@@ -280,4 +280,32 @@ input: Base64 strings to decode.
 output: Decoded strings.
 )doc");
 
+REGISTER_OP("Substr")
+  .Input("input: string")
+  .Input("pos: T")
+  .Input("len: T")
+  .Output("output: string")
+  .Attr("T: {int32, int64}")
+  .SetShapeFn(shape_inference::UnchangedShape)
```

Comment:
I think you can do a bit more here, validating that pos and len are both scalars:

```
.SetShapeFn([](InferenceContext* c) {
  ShapeHandle unused;
  TF_RETURN_IF_ERROR(c->WithRank(c->input(1), 0, &unused));
  TF_RETURN_IF_ERROR(c->WithRank(c->input(2), 0, &unused));
  return shape_inference::UnchangedShape(c);
}
```

If you do this, you might be able to just elide the shape checks in the kernel itself.

Of course, this depends on the discussion above about the shapes, so just FYI for now.